.github/workflows/compatibility-check-template.yml

```diff
@@ -75,21 +77,22 @@ jobs:
       - name: Check contracts using current branch
         working-directory: ./tools/compatibility-check
         run: |
+          go get github.com/onflow/cadence@${{ inputs.current-branch }}
+          go mod tidy
           go run ./cmd/check_contracts/main.go ../../tmp/contracts.csv ../../tmp/output-new.txt
 
       # Check contracts using base branch
 
-      - name: Checkout base branch
-        run: git checkout ${{ inputs.base-branch }}
-
```

Comment:
Instead of checking out the old branch, just do a cadence version bump on the tool